src/Microsoft.VisualStudio.Extensibility.Testing.SourceGenerator/TestServicesSourceGenerator.cs

```diff
@@ -1420,6 +1420,8 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                         }
                     }
 
+                    var isXunit3 = compilation.ReferencedAssemblyNames.Any(assemblyName => assemblyName.Name == "xunit.v3.assert");
```

Comment:
This isn't a good choice typically as a rule. Looking for a reference to `xunit.v3.core` is more likely correct, though in your case perhaps you don't have people using third party assertion libraries so it doesn't matter as much.